src/common/utils.cpp

```diff
@@ -206,26 +206,28 @@ unsigned get_jit_profiling_flags() {
     return flag;
 }
 
-static setting_t<std::string> jit_profiling_jitdumpdir;
-dnnl_status_t init_jit_profiling_jitdumpdir(
-        const char *jitdumpdir, bool overwrite) {
-#ifdef __linux__
-    static std::mutex m;
-    std::lock_guard<std::mutex> g(m);
-
-    if (jit_profiling_jitdumpdir.initialized() && !overwrite)
-        return status::success;
-
-    if (!jitdumpdir) {
+static std::string &jit_profiling_jitdumpdir() {
+    // initialize on first use with default values from environmenet
+    static std::string s = []() -> std::string {
         char buf[PATH_MAX];
```

Comment:
`PATH_MAX` used to be under `__linux__` macro as it's a linux specific macro. Looks like `jit_profiling_jitdumpdir()` function should be under `__linux___` macro as it's used only on linux.